src/lib/polygon/PolygonKey.js

```diff
@@ -0,0 +1,145 @@
+/* global Nimiq */
+/* global Errors */
+/* global ethers */
+/* global CONFIG */
+
+class PolygonKey { // eslint-disable-line no-unused-vars
+    /**
+     * @param {Key} key
+     */
+    constructor(key) {
+        if (key.type !== Nimiq.Secret.Type.ENTROPY) {
+            throw new Errors.KeyguardError('Key must be of type Nimiq.Entropy');
+        }
+        this._key = key;
+    }
+
+    /**
+     * @param {string} path
+     * @returns {string}
+     */
+    deriveAddress(path) {
+        const wallet = this.deriveKeyPair(path);
+        return wallet.address;
+    }
+
+    /**
+     * @param {string} path
+     * @param {ethers.providers.TransactionRequest} transaction
+     * @returns {Promise<string>}
+     */
+    async sign(path, transaction) {
+        const wallet = this.deriveKeyPair(path);
+        return wallet.signTransaction(transaction);
+    }
+
+    /**
+     *
+     * @param {string} path
+     * @param {ethers.TypedDataDomain} domain
+     * @param {Record<string, Array<ethers.TypedDataField>>} types
+     * @param {Record<string, any>} value
+     * @returns {Promise<string>}
+     */
+    async signTypedData(path, domain, types, value) {
+        const wallet = this.deriveKeyPair(path);
+        return wallet._signTypedData(domain, types, value);
+    }
+
+    /**
+     * @param {string} path
+     * @param {ethers.Contract} usdcContract
```

Comment:
I did it this way because creating the contract in the `PolygonKey.js` file means adding a dependency on the `PolygonContractABIs.js` file, which would then be required for all places where a `PolygonKey` is created, but not used to sign, like Create/Import/DerivePolygonAddress, where the additional bundled ABIs are only dead weight.